src/shared/managers/PermissionManager.ts

```diff
@@ -26,47 +21,15 @@ export default class PermissionManager {
       );
     }
 
-    return await OneSignal.context.permissionManager.getNotificationPermission(
-      OneSignal.config!.safariWebId,
-    );
+    return await OneSignal.context.permissionManager.getNotificationPermission();
   }
 
   /**
-   * Notification permission reported by the browser.
-   *
-   * @param safariWebId The Safari web ID necessary to access the permission
-   * state on Legacy Safari on macOS.
-   */
-  public async getNotificationPermission(
-    safariWebId?: string,
-  ): Promise<NotificationPermission> {
-    if (Environment.useSafariLegacyPush()) {
-      return PermissionManager.getLegacySafariNotificationPermission(
-        safariWebId,
-      );
-    }
-    return this.getW3cNotificationPermission();
-  }
-
-  /**
-   * Returns the Safari browser's notification permission as reported by the browser.
-   *
-   * @param safariWebId The Safari web ID necessary for Legacy Safari on macOS.
-   */
-  private static getLegacySafariNotificationPermission(
-    safariWebId?: string,
-  ): NotificationPermission {
-    if (safariWebId)
-      return window.safari.pushNotification.permission(safariWebId)
-        .permission as NotificationPermission;
```

Comment:
We need to check if `window.safari.pushNotification.permission(safariWebId)` returns false, and only then use the new VAPID push. This is due to a bug that still exists today in Safari that doesn't allow migrating someone who has accept Safari legacy push for the domain to standard notifications.

We this Asana ticket for more details:
https://app.asana.com/1/780103692902078/project/1204295426001934/task/1204303093244836?focus=true